src/metabase/query_processor/middleware/annotate.clj

```diff
@@ -496,17 +520,20 @@
 (defn- flow-field-metadata
   "Merge information about fields from `source-metadata` into the returned `cols`."
   [source-metadata cols]
-  (let [field-id->metadata (u/key-by :id source-metadata)]
+  (let [ref->metadata (u/key-by (comp u/field-ref->key :field_ref) source-metadata)]
     (for [col cols]
-      (if-let [source-metadata-for-field (-> col :id field-id->metadata)]
+      (if-let [source-metadata-for-field (-> col :field_ref u/field-ref->key ref->metadata)]
         (merge-source-metadata-col source-metadata-for-field col)
         col))))
 
+(declare mbql-cols)
```

Comment:
why not just move the definition of `mbql-cols` to here?

Reply:
`cols-for-source-query` and `mbql-cols` are mutually recursive. one has to be declared. I think this behavior is a carry-over from before and not new now. Might be in the diff due to reordering of other things though.